src/devices/Amg88xx/tests/Amg88xxTests.cs

```diff
@@ -0,0 +1,629 @@
+            Assert.Equal(temperatureOverflow, status);
+        }
+
+        [Fact]
+        public void ClearTemperatureOverflowTest()
+        {
+            I2cTestDevice i2cDevice = new I2cTestDevice();
+            Amg88xx sensor = new Amg88xx(i2cDevice);
+
+            sensor.ClearTemperatureOverflow();
+
+            Assert.Equal(2, i2cDevice.DataWritten.Count);
+            Assert.Equal((byte)Register.SCLR, i2cDevice.DataWritten.Dequeue());
+            Assert.Equal(1 << (byte)StatusClearBit.OVFCLR, i2cDevice.DataWritten.Dequeue());
+        }
+
+        [Theory]
+        [InlineData(1 << (byte)StatusFlagBit.OVF_THS, true)]
+        [InlineData(0xff ^ (1 << (byte)StatusFlagBit.OVF_THS), false)]
+        public void HasThermistorOverflowTest(byte statusRegisterContent, bool thermistorOverflow)
+        {
+            I2cTestDevice i2cDevice = new I2cTestDevice();
+            Amg88xx sensor = new Amg88xx(i2cDevice);
+
+            i2cDevice.DataToRead.Enqueue(statusRegisterContent);
+
+            bool status = sensor.HasThermistorOverflow();
+
+            Assert.Single(i2cDevice.DataWritten);
+            Assert.Equal((byte)Register.STAT, i2cDevice.DataWritten.Dequeue());
+            Assert.Equal(thermistorOverflow, status);
+        }
+
+        [Fact]
+        public void ClearThermistorOverflowTest()
+        {
+            I2cTestDevice i2cDevice = new I2cTestDevice();
+            Amg88xx sensor = new Amg88xx(i2cDevice);
+
+            sensor.ClearThermistorOverflow();
+
+            Assert.Equal(2, i2cDevice.DataWritten.Count);
+            Assert.Equal((byte)Register.SCLR, i2cDevice.DataWritten.Dequeue());
+            Assert.Equal(1 << (byte)StatusClearBit.OVFTHCLR, i2cDevice.DataWritten.Dequeue());
+        }
+
+        [Theory]
+        [InlineData(1 << (byte)StatusFlagBit.INTF, true)]
+        [InlineData(0xff ^ (1 << (byte)StatusFlagBit.INTF), false)]
+        public void HasInterruptTest(byte statusRegisterContent, bool interrupt)
+        {
+            I2cTestDevice i2cDevice = new I2cTestDevice();
+            Amg88xx sensor = new Amg88xx(i2cDevice);
+
+            i2cDevice.DataToRead.Enqueue(statusRegisterContent);
+
+            bool status = sensor.HasInterrupt();
+
+            Assert.Single(i2cDevice.DataWritten);
+            Assert.Equal((byte)Register.STAT, i2cDevice.DataWritten.Dequeue());
+            Assert.Equal(interrupt, status);
+        }
+
+        [Fact]
+        public void ClearInterruptTest()
+        {
+            I2cTestDevice i2cDevice = new I2cTestDevice();
+            Amg88xx sensor = new Amg88xx(i2cDevice);
+
+            sensor.ClearInterrupt();
+
+            Assert.Equal(2, i2cDevice.DataWritten.Count);
+            Assert.Equal((byte)Register.SCLR, i2cDevice.DataWritten.Dequeue());
+            Assert.Equal(1 << (byte)StatusClearBit.INTCLR, i2cDevice.DataWritten.Dequeue());
+        }
+
+        [Fact]
+        public void ClearAllFlagsTest()
```

Comment:
i.e. if APIs were designed a bit differently you could test for i.e.:
`OverflowTriggeredARetry`, this way user don't need to know about overflow at all (treat this as an example you might have a better idea about how to use this properly)

Reply:
@krwq As stated in my comment for HasTemperatureOverflow. If there is any use for the overflow indication it would be up to the user to deal with it. The sensor will still work as usual, if the indicator is not reset. It doesn't require a retry. It is a kind of "history" information that can be evaluated by the application. 
I don't want to make any assumptions about its actual use as I don't know the requirements of the sensor users.